### PR TITLE
[improve][broker] Add callback parameters to the SendCallback.sendComplete

### PR DESCRIPTION
**Pull request by crossoverJie**
PIP: #22940 

### Modifications

- Add `OpSendMsgStats` parameters to the `org.apache.pulsar.client.impl.SendCallback.sendComplete`

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/crossoverJie/pulsar/pull/29

### CODE REVIEW

**Review by dao-jun — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ProduceWithMessageIdTest.java [hunk 1]
```diff
@@ -115,4 +125,70 @@ public CompletableFuture<MessageId> getFuture() {
...
+            MessageMetadata metadata = new MessageMetadata();
+            ByteBuffer buffer = ByteBuffer.wrap("data".getBytes(StandardCharsets.UTF_8));
+            MessageImpl<byte[]> msg = MessageImpl.create(metadata, buffer, Schema.BYTES, topic);
+            msg.getDataBuffer().retain();
+            totalReadabled += msg.getDataBuffer().readableBytes();
+            totalUncompressedSize += msg.getUncompressedSize();
+            producer.sendAsync(msg, sendComplete);
+        }
+
+        Awaitility.await().untilTrue(result);
```
Comment (RobertIndie) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ProduceWithMessageIdTest.java:183]: We can use CountDownLatch watch here.

**Review by RobertIndie — APPROVED** [review]

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**